clambcc/clambc-compiler.py

```diff
@@ -491,11 +588,124 @@ def createInputSourceFile(clangLLVM: ClangLLVM, name: str, args: list, options:
+        , "__clambc_virusnames"
+        , "__clambc_filesize"
+        , "__clambc_match_counts"
+        , "__clambc_match_offsets"
+        , "__clambc_pedata"
+        , "__Copyright" 
+        ]
+
+OPTIMIZE_OPTIONS = ["-S"
+        , "--disable-loop-unrolling"
+        , " --disable-i2p-p2i-opt"
+        , " --disable-loop-unrolling"
+        , " --disable-promote-alloca-to-lds"
+        , " --disable-promote-alloca-to-vector"
+        , " --disable-simplify-libcalls"
+        , " --disable-tail-calls"
+        , " --vectorize-slp=false"
+        , " --vectorize-loops=false"
+        , " -internalize-public-api-list=\"%s\"" % ','.join(INTERNALIZE_API_LIST)
+        ]
+
+#TODO: Remove this when we properly handle opaque pointers.
+OPTIMIZE_OPTIONS.append("-opaque-pointers=0")
+
+OPTIMIZE_PASSES = ["function(mem2reg)"
+        , 'verify'
+        , 'clambc-remove-undefs' #TODO: MAY NOT BE NEEDED 
+        , 'verify'
+        , 'clambc-preserve-abis'
+        , 'verify'
+        , 'default<O3>'
+        , 'globalopt'
+        , 'clambc-preserve-abis' #remove fake function calls because O3 has already run
+        , 'verify'
+        , 'clambc-remove-unsupported-icmp-intrinsics'
+        , 'verify'
+        , 'clambc-remove-usub'
+        , 'verify'
+        , 'clambc-remove-fshl'
+        , 'verify'
+        , 'clambc-lowering-notfinal' # perform lowering pass
+        , 'verify'
+        , 'lowerswitch'
+        , 'verify'
+        , 'clambc-remove-icmp-sle'
+        , 'verify'
+        , 'function(clambc-verifier)'
+        , 'verify'
+        , 'clambc-remove-freeze-insts'
+        , 'verify'
+        , 'clambc-lowering-notfinal'  # perform lowering pass
+        , 'verify'
+        , 'clambc-lcompiler-helper' #compile the logical_trigger function to a
+        , 'verify'
+        , 'clambc-lcompiler' #compile the logical_trigger function to a
+        , 'verify'
+        , 'internalize'
+        , 'verify'
+        , 'clambc-rebuild'
+        , 'verify'
+        , 'clambc-trace'
+        , 'verify'
+        , 'clambc-outline-endianness-calls'
+        , 'verify'
+        , 'clambc-change-malloc-arg-size' #TODO: MAY NOT BE NEEDED 
+        , 'verify'
+        , 'clambc-extend-phis-to-64-bit'
+        , 'verify'
+        , 'clambc-convert-memsets-to-32Bit'
+        , 'verify'
+        , 'globalopt'
+        , 'clambc-prepare-geps-for-writer'
+        , 'verify'
+        , 'clambc-writer'
+        , 'verify'
+]
+
+OPTIMIZE_LOADS=[ f"--load {SHARED_OBJ_DIR}/libclambccommon.so"
+        , f"--load-pass-plugin {SHARED_OBJ_DIR}/libclambcremoveundefs.so"
+        , f"--load-pass-plugin {SHARED_OBJ_DIR}/libclambcpreserveabis.so"
+        , f"--load-pass-plugin {SHARED_OBJ_DIR}/libclambcremoveunsupportedicmpintrinsics.so"
+        , f"--load-pass-plugin {SHARED_OBJ_DIR}/libclambcremoveusub.so"
+        , f"--load-pass-plugin {SHARED_OBJ_DIR}/libclambcremovefshl.so"
+#        , f"--load-pass-plugin {SHARED_OBJ_DIR}/libclambcremovepointerphis.so"
```

Comment:
Should this be commented out? I see there is indeed a pass for this.

Reply:
That pass appears to no longer be necessary.  It was needed at the time of the previous release, but that may have been fixed by the upgrades to the runtime.  I am not sure, but it appears to no longer be needed now.  I was going to wait to delete the pass (and these lines) until after the -rc period, in case something comes up where I need to put it back.